rules.d/50-udev-default.rules.in

```diff
@@ -89,4 +89,7 @@ KERNEL=="udmabuf", GROUP="kvm"
 
 SUBSYSTEM=="ptp", ATTR{clock_name}=="KVM virtual PTP", SYMLINK += "ptp_kvm"
 
+SUBSYSTEM=="dmi", ATTR{sys_vendor}!="LENOVO", ENV{ID_VENDOR}="$attr{sys_vendor}", ENV{ID_MODEL}="$attr{product_name}"
+SUBSYSTEM=="dmi", ATTR{sys_vendor}=="LENOVO", ENV{ID_VENDOR}="$attr{sys_vendor}", ENV{ID_MODEL}="$attr{product_version}"
```

Comment:
So the goal really should be to keep the vendor matches out of the generic rule sets. One approach here would be to store the dmi field to use in the hwdb. i.e. introduce a hwdb property ID_SYSFS_ATTRIBUTE_MODEL or so which we'd set to `produc_version` for everything matching lenovo. Then, just place a udev rule here that checks if that property is set, and if so, reads the model from that attribute, and otherwise uses `product_name`.

That way we keep everything vendor-specific in the hwdb and the udev rules are entirely generic.